packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -497,7 +510,26 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         metered(_gasLimit)
     {
         (address token,) = gasPayingToken();
-        if (token != Constants.ETHER && msg.value != 0) revert NoValue();
+        address l2ETHToken = l2ETHToken();
+        if (token != Constants.ETHER && msg.value != 0 && l2ETHToken == address(0)) revert NoValue();
+        if (token != Constants.ETHER && msg.value != 0 && l2ETHToken != address(0)) {
+            require(
```

Comment:
We can combine the initial check for `token != Constants.ETHER` and `msg.value != 0` into a single condition.

Reply:
Not sure how to make changes. Could you please clarify this?